File: seed/php-sdk/trace/src/V2/V3/Problem/Types/GetFunctionSignatureRequest.php
<?php

namespace Seed\V2\V3\Problem\Types;

use Seed\Core\Json\JsonSerializableType;
use Seed\Core\Json\JsonProperty;

class GetFunctionSignatureRequest extends JsonSerializableType
{
    /**
     * @var mixed $functionSignature
     */
    #[JsonProperty('functionSignature')]
    public mixed $functionSignature;

    /**
     * @param array{
     *   functionSignature: mixed,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->functionSignature = $values['functionSignature'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/problem/types/get_function_signature_request.rb
# frozen_string_literal: true

module Seed
  module V2
    module Problem
      module Types
        class GetFunctionSignatureRequest < Internal::Types::Model
          field :function_signature, lambda {
            Seed::V2::Problem::Types::FunctionSignature
          }, optional: false, nullable: false
        end
      end
    end
  end
end
